PowerShellTools/ServiceManagement/PowershellHostProcessHelper.cs

```diff
@@ -61,34 +106,94 @@ public static PowershellHostProcess CreatePowershellHostProcess()
             bool success = readyEvent.WaitOne();
             readyEvent.Close();
 
-            MakeTopMost(powershellHostProcess.MainWindowHandle);
+            MakeTopMost(_powerShellHostProcess.MainWindowHandle);
 
-            #if !DEBUG
-                ShowWindow(powershellHostProcess.MainWindowHandle, SW_HIDE);
-            #endif
+            ShowWindow(_powerShellHostProcess.MainWindowHandle, SW_HIDE);
 
             if (!success)
             {
-                int processId = powershellHostProcess.Id;
+                int processId = _powerShellHostProcess.Id;
                 try
                 {
-                    powershellHostProcess.Kill();
+                    _powerShellHostProcess.Kill();
                 }
                 catch (Exception)
                 {
                 }
 
-                if (powershellHostProcess != null)
+                if (_powerShellHostProcess != null)
                 {
-                    powershellHostProcess.Dispose();
-                    powershellHostProcess = null;
+                    _powerShellHostProcess.Dispose();
+                    _powerShellHostProcess = null;
                 }
                 throw new PowershellHostProcessException(String.Format(CultureInfo.CurrentCulture,
                                                                         Resources.ErrorFailToCreateProcess,
                                                                         processId.ToString()));
             }
 
-            return new PowershellHostProcess(powershellHostProcess, endPointGuid);
+            _inputStreamWriter = _powerShellHostProcess.StandardInput;
+
+            return new PowerShellHostProcess(_powerShellHostProcess, EndPointGuid);
+        }
+
+        /// <summary>
+        /// Monitoring thread for user input request
+        /// </summary>
+        /// <remarks>
+        /// Will be started once app begins to run on remote PowerShell host service
+        /// Stopped once app exits
+        /// </remarks>
+        public static void MonitorUserInputRequest()
+        {
+            while (AppRunning)
+            {
+                foreach (ProcessThread thread in _powerShellHostProcess.Threads)
+                {
+                    if (thread.ThreadState == System.Diagnostics.ThreadState.Wait
+                        && thread.WaitReason == ThreadWaitReason.UserRequest)
+                    {
+                        if (PowerShellToolsPackage.Debugger != null &&
+                            PowerShellToolsPackage.Debugger.HostUi != null)
+                        {
+                            string inputText = PowerShellToolsPackage.Debugger.HostUi.ReadLine(Resources.UserInputRequestMessage);
+
+                            // Feed into stdin stream
+                            _inputStreamWriter.WriteLine(inputText);
+                            break;
+                        }
+                    }
+                }
+
+                Thread.Sleep(1000); 

```

Comment:
This is running in background, so no one gets freeze till we pop up dialog on UI thread, however I changed to give 500ms interval for polling on input so that it respond faster. And I feels like by simply sleeping should be good enough for this case, adding extra timer/callback is a bit overkill in my opinion.